### PR TITLE
Rewrite simplify type names using SemanticModel action and a tree walker

### PR DESCRIPTION
**Pull request by sharwell**
This is a structural reorganization which allows future improvements to be implemented with less code churn. Extracted from #40746.

### CODE REVIEW

File: src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            _analyzer = analyzer;
+            _semanticModel = semanticModel;
+            _optionSet = optionSet;
+            _cancellationToken = cancellationToken;
+        }
+
+        public override void VisitQualifiedName(QualifiedNameSyntax node)
+        {
+            if (node.IsKind(SyntaxKind.QualifiedName) && TrySimplify(node))
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs]: the IsKind check isn't necessary, right?

File: src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                return;
+            }
+
+            // descend further.
+            DefaultVisit(node);
+        }
+
+        public override void VisitMemberAccessExpression(MemberAccessExpressionSyntax node)
+        {
+            if (node.IsKind(SyntaxKind.SimpleMemberAccessExpression) && TrySimplify(node))
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs]: this is the only Kind check that seems necessayr IMO.

File: src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs (hunk 1 above)
Comment (sharwell) [reply src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs]: I believe some of the tests can be removed, but I wanted the initial refactoring to be explicit.

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs]: k :)

**Comment by CyrusNajmabadi** [comment]
@dpoeschl @sharwell want to restart the linux run?  failed with:

```
Downloading 'https://dot.net/v1/dotnet-install.sh'
curl: (35) gnutls_handshake() failed: The TLS connection was non-properly terminated.
##[error](NETCORE_ENGINEERING_TELEMETRY=InitializeToolset) Failed to acquire dotnet install script (exit code '35').
Killing running build processes...
```

**Pull request merged**